easybuild/easyconfigs/c/Charm++/Charm++-gerrit1886-CrayGNU-2016.11.eb

```diff
@@ -0,0 +1,31 @@
+easyblock = 'CmdCp'
+
+name = 'Charm++'
+version = 'gerrit1886'
+
+homepage = 'http://charm.cs.illinois.edu/research/charm/'
+description = """Charm++ is a machine independent parallel programming system. Programs written using this system will run unchanged on MIMD machines with or without a shared memory."""
+
+toolchain = {'name': 'CrayGNU', 'version': '2016.11'}
+toolchainopts = {'dynamic': False}
```

Comment:
If you are using cuda, I think you can remove this line: it will be dynamic by default. 
BTW, that would explain why your build fails without the patch: this ugni/pmi problem occurs only when linking it dynamically.